app/controllers/application_controller.rb

```diff
@@ -3,5 +3,9 @@
 class ApplicationController < ActionController::Base
   # Only allow modern browsers supporting webp images, web push, badges, import maps, CSS nesting, and CSS :has.
   allow_browser versions: :modern
+
+  def authenticate_admin!
+    redirect_to root_path unless current_user&.is_admin?
```

Comment:
The `&` in `current_user&.is_admin?` is like saying `defined?(current_user) && current_user.is_admin?` ;)